src/operator/tensor/dot-inl.cuh

```diff
@@ -442,6 +442,99 @@ struct DotCsrRspDnsScalarKernel {
   }
 };
 
+/*!
+ * \brief GPU Kernel to re-arrange nnz elements to csc order
+ * Parallelization by output elements: 1 thread/row of csr
+ */
+struct CscDataIndicesKernel {
+  template<typename DType, typename IType, typename CType>
+  __device__ __forceinline__ static void Map(int tid,
+                                             const DType* csr_data,
+                                             const IType* csr_indices,
+                                             const CType* csr_indptr,
+                                             DType* csc_data,
+                                             unsigned long long* csc_indices,
+                                             unsigned long long* csc_indptr,
+                                             unsigned long long* workspace,
```

Comment:
Rename "workspace" to "col_counters"? Better document what this space is used for:
e.g. used to count the offset of column indices atomically

Reply:
Done